### PR TITLE
add subaru outback/legacy to subaru safety

### PR DESCRIPTION
**Pull request by bugsy924**

### CODE REVIEW

**Comment by bugsy924** [comment]
I'll leave this open for now. We're aware of the failed Misra check

**Comment by bugsy924** [comment]
Line 90 (now 92 and passes) failed Misra check 
Other safeties seem to have the version that failed. Should they be changed?

File: board/safety/safety_subaru.h [hunk 1]
```diff
@@ -44,9 +45,10 @@ static int subaru_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
   int addr = GET_ADDR(to_send);
 
   // steer cmd checks
-  if (addr == 0x122) {
-    int desired_torque = ((GET_BYTES_04(to_send) >> 16) & 0x1FFF);
-    bool violation = 0;
+  if ((addr == 0x122) || (addr == 0x164)) {
+    int bit_shift = (addr == 0x122) ? 16 : 8;
+    int desired_torque = ((GET_BYTES_04(to_send) >> bit_shift) & 0x1FFF);
+    int violation = 0;
```
Comment (rbiasini) [inline board/safety/safety_subaru.h]: keep violation a bool

File: board/safety/safety_subaru.h [hunk 2]
```diff
@@ -87,7 +89,7 @@ static int subaru_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
       subaru_ts_last = ts;
     }
 
-    if (violation) {
+    if (violation != 0) {
```
Comment (rbiasini) [inline board/safety/safety_subaru.h]: violation can remain a bool, so this change isn't needed

File: board/safety/safety_subaru.h [hunk 3]
```diff
@@ -100,10 +102,10 @@ static int subaru_fwd_hook(int bus_num, CAN_FIFOMailBox_TypeDef *to_fwd) {
   int bus_fwd = -1;
   if (bus_num == 0) {
     bus_fwd = 2;  // Camera CAN
-  }
+  }
```
Comment (rbiasini) [inline board/safety/safety_subaru.h]: space

**Comment by bugsy924** [comment]
Can this be merged?

**Pull request merged**